### PR TITLE
Decoration config supports setting memory limit equal to request

### PR DESCRIPTION
**Pull request by chaodaiG**
This will help solving noisy neighbor problem by enforcing pod resource limit equal to request, this can be set per org/repo/cluster. The build cluster itself could choose to set limit-range to cover cases where request is not set on a job

/cc @cjwagner @alvaroaleman

### CODE REVIEW

**Comment by alvaroaleman** [comment]
At least for CPU it is not recommended to set limits ever, as it will always throttle the workload even if there is sufficient CPU available (sorry, I don't have time to find a source for this right now). Maybe just do this for mem? Or configurable?

**Comment by chaodaiG** [comment]
> At least for CPU it is not recommended to set limits ever, as it will always throttle the workload even if there is sufficient CPU available (sorry, I don't have time to find a source for this right now). Maybe just do this for mem? Or configurable?

I agree with you, had similar experience and I know the difference between cpu and mem.

**Comment by chaodaiG** [comment]
updated

**Comment by chaodaiG** [comment]
@alvaroaleman , mind take another look?

**Comment by chaodaiG** [comment]
ping for review

File: prow/pod-utils/decorate/podspec.go
Comment (cjwagner) [inline prow/pod-utils/decorate/podspec.go:251]: I think we should do this with the rest of the decoration in the `decorate()` function.

**Review by cjwagner — APPROVED** [review]
I have one nit, otherwise LGTM.
/hold

File: prow/pod-utils/decorate/podspec.go
Comment (chaodaiG) [reply prow/pod-utils/decorate/podspec.go]: interesting enough that I thought I did it on `decorate()` method, fixed now

File: prow/pod-utils/decorate/podspec.go
Comment (cjwagner) [inline prow/pod-utils/decorate/podspec.go:819]: `pj.Spec.DecorationConfig != nil` can be assumed here. It doesn't make a difference though.

**Review by cjwagner — APPROVED** [review]

**Pull request merged**
